src/VirtualClient/VirtualClient.Contracts/ParallelLoopExecution.cs

```
@@ -74,26 +74,28 @@ private async Task ExecuteComponentLoopAsync(VirtualClientComponent component, E
-                }
-                else
-                {
-                    this.Logger.LogMessage($"{nameof(ParallelLoopExecution)} Component = {component.TypeName}", LogLevel.Information, telemetryContext);
-                }
+                    string scenarioMessage = string.IsNullOrWhiteSpace(component.Scenario)
+                    ? $"{nameof(ParallelLoopExecution)} Component = {component.TypeName}"
+                    : $"{nameof(ParallelLoopExecution)} Component = {component.TypeName} (scenario={component.Scenario})";
 
-                // Create a task to execute the component, respecting the timeout.
-                Task componentExecutionTask = component.ExecuteAsync(cancellationToken);
+                    this.Logger.LogMessage(scenarioMessage, LogLevel.Information, telemetryContext);
 
-                // Wait for the task to complete or timeout, also determining which task completed.
-                if (await Task.WhenAny(componentExecutionTask, this.timeoutTask) == componentExecutionTask)
-                {
-                    // The component has finished execution, restart the loop.
-                    continue;
+                    // Execute the component task with timeout handling.
+                    Task componentExecutionTask = component.ExecuteAsync(cancellationToken);
+                    Task completedTask = await Task.WhenAny(componentExecutionTask, this.timeoutTask);
+
+                    if (completedTask == this.timeoutTask)
+                    {
+                        break;
+                    }
+
+                    await componentExecutionTask;
                 }
-                else
+                catch (Exception ex)
                 {
+                    this.Logger.LogMessage($"{nameof(ParallelLoopExecution)} Component = {component.TypeName} encountered an error: {ex.Message}", LogLevel.Error, telemetryContext);
```

Instead of just logging, should we throw a non-fatal exception like "WorkloadException" if any of the workload fails? This would make sure that workload related errors are not missed or treated as false positives?

---

Ack.